tests/testthat/test-filter.R

```diff
@@ -5,71 +5,70 @@ lessthan <- function(...) {}
 morethan <- function(...) {}
 interval <- function(...) {}
 
-# TODO: implement filters, check number of samples after filtering the test data and fill expected_samples
 test_cases <- list(
   list(
     method = "by.meta", query = list(Status = include("C")),
-    expected_samples = 0
+    expected_samples = 6
   ),
   list(
-    method = "by.meta", method = "by.meta", query = list(Lane = exclude("A")),
-    expected_samples = 0
+    method = "by.meta", query = list(Lane = exclude("A")),
+    expected_samples = 8
   ),
   list(
     method = "by.meta", query = list(Age = lessthan(23)),
-    expected_samples = 0
+    expected_samples = 10
   ),
   list(
     method = "by.meta", query = list(Age = interval(15, 23)),
-    expected_samples = 0
+    expected_samples = 4
   ),
   list(
     method = "by.meta", query = list(Lane = include("B", "C")),
-    expected_samples = 0
+    expected_samples = 8
   ),
   list(
     method = "by.meta", query = list(Lane = exclude("A", "C")),
-    expected_samples = 0
+    expected_samples = 2
   ),
   list(
     method = "by.meta", query = list(Lane = include("A", "B"), Age = morethan(15)),
-    expected_samples = 0
+    expected_samples = 3
   ),
   list(
-    method = "by.repertoire", query = list(n_clonotypes = morethan(1000)),
-    expected_samples = 0
+    method = "by.repertoire", query = list(n_clonotypes = morethan(6000)),
+    expected_samples = 8
   ),
   list(
     method = "by.repertoire", query = list(n_clones = lessthan(100)),
     expected_samples = 0
   ),
   list(
-    method = "by.rep", query = list(n_clonotypes = morethan(1000)),
-    expected_samples = 0
+    method = "by.rep", query = list(n_clonotypes = morethan(6000)),
+    expected_samples = 8
   ),
   list(
     method = "by.rep", query = list(n_clones = lessthan(100)),
     expected_samples = 0
   ),
   list(
     method = "by.clonotype", query = list(CDR3.aa = exclude("partial", "out_of_frame")),
-    expected_samples = 0
+    expected_samples = 12
   ),
   list(
-    method = "by.clonotype", query = list(Clones = interval(5, 50)),
-    expected_samples = 0
+    method = "by.clonotype", query = list(Clones = interval(150, 200)),
+    expected_samples = 4
   ),
   list(
     method = "by.cl", query = list(CDR3.aa = exclude("partial", "out_of_frame")),
-    expected_samples = 0
+    expected_samples = 12
   ),
   list(
-    method = "by.cl", query = list(Clones = interval(5, 50)),
-    expected_samples = 0
+    method = "by.cl", query = list(Clones = interval(150, 200)),
+    expected_samples = 4
   ),
   list(
     method = "by.gene", query = list(V = exclude("TRBV1", "TRGV11")),
```

Comment:
Maybe I am misunderstanding this point, but in my mind, by.gene should include or exclude clonotypes with target gene so the user will have each sample without clonotypes of target genes or alleles (for example, I am interested in segment-level analysis without some specific alleles); @tsvvas can you resolve this point?

Reply:
```

method = "by.gene", query = list(V = exclude("TRBV1", "TRGV11")), match="exact"
method = "by.gene", query = list(V = exclude("TRBV1", "TRGV11")), match="substring"
method = "by.gene", query = list(V = exclude("TRBV1", "TRGV11")), match="startswith"
method = "by.gene", query = list(V = exclude("TRBV1", "TRGV11")), match="regexp"

```

Reply:
`by.gene` is redundant since we already have `by.clonotype`